src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/frequency/FrequencyBoardOrAlightEvent.java

```diff
@@ -110,7 +106,7 @@ public RaptorTripPattern pattern() {
 
   @Override
   public int transitReluctanceFactorIndex() {
-    return pattern.getMode().ordinal();
+    return raptorTripPattern.getTripPattern().getPattern().getMode().ordinal();
```

Comment:
I think we should cache the `modeOrdinal` in each raptorTripPattern - doing this traversal will fetch the objects from memory. I think this slows down the search.

Reply:
Should I just use it as is, or should it be extracted to a separate method, like how [`SlackProvider` is used in `TripPatternBuilder`](https://github.com/opentripplanner/OpenTripPlanner/blob/296a0dd3ee0d7b5e4df6c12969dcb5bb145cb953/src/main/java/org/opentripplanner/transit/model/network/TripPatternBuilder.java#L77-L80)?

Reply:
Maybe not so important, the reuse of the static method enforce the index to be generated the same way and is inlined - fast. I think we will refactor this anyway to support other use-cases/merge it with the (un)preferred stuff - so make is simple for now.

Reply:
Added caching of this in RoutingTripPattern